tsl/src/nodes/skip_scan/planner.c

```diff
@@ -422,51 +479,71 @@ get_distinct_var(PlannerInfo *root, IndexPath *index_path, SkipScanPath *skip_sc
 
 	RangeTblEntry *ht_rte = planner_rt_fetch(var->varno, root);
 	RangeTblEntry *chunk_rte = planner_rt_fetch(rel->relid, root);
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_rte->relid, false);
```

Comment:
These lookups are expensive, please use the `cached_chunk_struct` member of `fdw_private` instead.

Reply:
Right I am aware of it. If you remember I discussed it during the standup to add chunk information to TimescaleDBPrivate. It wasn't there couple of months back :-)
Good to see that Chunk Struct is cached. If I remember correctly, its not propagating to compressed hypertable children (I maybe  wrong on this) that's why I deferred it to my TODO list.

If you don't insist,  
Lets park it for next phase.

Reply:
I verified it and its not propagating to compressed chunks hierarchy. 
```
(gdb) p (RelOptInfo*) root->simple_rel_array[0]
$8 = (RelOptInfo *) 0x0
(gdb) p (RelOptInfo*) root->simple_rel_array[1]
$9 = (RelOptInfo *) 0x5640c4f6e668
(gdb) p (TimescaleDBPrivate*)$9->fdw_private 
$10 = (TimescaleDBPrivate *) 0x5640c4f6e958
(gdb) p *(TimescaleDBPrivate*)$9->fdw_private 
$11 = {appends_ordered = false, order_attno = 0, nested_oids = 0x0, chunk_oids = 0x0, serverids = 0x0, server_relids = 0x0, fdw_relation_info = 0x0, 
  cached_chunk_struct = 0x0, compressed_ec_em_pairs = 0x0}
(gdb) p (RelOptInfo*) root->simple_rel_array[2]
$12 = (RelOptInfo *) 0x5640c4fe8008
(gdb) p *(TimescaleDBPrivate*)$12->fdw_private 
$13 = {appends_ordered = false, order_attno = 0, nested_oids = 0x0, chunk_oids = 0x0, serverids = 0x0, server_relids = 0x0, fdw_relation_info = 0x0, 
  cached_chunk_struct = 0x5640c4f6db68, compressed_ec_em_pairs = 0x0}
(gdb) p (RelOptInfo*) root->simple_rel_array[3]
$14 = (RelOptInfo *) 0x5640c4fe9380
(gdb) p *(TimescaleDBPrivate*)$14->fdw_private 
$15 = {appends_ordered = false, order_attno = 0, nested_oids = 0x0, chunk_oids = 0x0, serverids = 0x0, server_relids = 0x0, fdw_relation_info = 0x0, 
  cached_chunk_struct = 0x5640c50197d0, compressed_ec_em_pairs = 0x0}
(gdb) p (RelOptInfo*) root->simple_rel_array[4]
$16 = (RelOptInfo *) 0x5640c4fe99d0
(gdb) p *(TimescaleDBPrivate*)$16->fdw_private 
$17 = {appends_ordered = false, order_attno = 0, nested_oids = 0x0, chunk_oids = 0x0, serverids = 0x0, server_relids = 0x0, fdw_relation_info = 0x0, 
  cached_chunk_struct = 0x0, compressed_ec_em_pairs = 0x5640c5019328}
(gdb) 

```

Reply:
created an issue to implement it.
https://github.com/timescale/timescaledb/issues/6064

Reply:
> created an issue to implement it.
> #6064

We shouldn't do this, it will be a planning time regression. We don't fill up the entire Chunk struct for compressed chunks on purpose, to save time.

Probably you'll have to add parameters to `skip_scan_path_create` and `get_distinct_var` to do varattno translation differently, based on whether the input path is compressed. This is known at the level of `build_subpaths` where `skip_scan_path_create` is called.